testing/test_cache.py

```diff
@@ -87,7 +87,35 @@ def test_cachefuncarg(cache):
         assert result.ret == 0
         result.stdout.fnmatch_lines(["*1 passed*"])
 
-
+    def test_custom_rel_cache_dir(self, testdir):
+        rel_cache_dir = os.path.join('custom_cache_dir', 'subdir')
+        testdir.makeini("""
+            [pytest]
+            cache_dir = {cache_dir}
+        """.format(cache_dir=rel_cache_dir))
+        testdir.makepyfile(test_errored='def test_error():\n    assert False')
+        testdir.runpytest()
+        assert testdir.tmpdir.join(rel_cache_dir).isdir()
+
+    def test_custom_abs_cache_dir(self, testdir):
+        abs_cache_dir = os.path.abspath(testdir.tmpdir.join('custom_cache_dir').strpath)
```

Comment:
To ensure we are using a non-sibling of our `tmpdir`, please use the `tmpdir_factory` fixture to create a new temporary directory instead:

```python
def test_custom_abs_cache_dir(self, testdir, tmpdir_factory):
    abs_cache_dir = str(tmpdir_factory.mktemp('custom_cache_dir'))
```

Better to use a non-sibling directory to avoid this feature breaking in the future by accident.